diff --git a/src/Gui/PreferencePages/DlgSettingsNavigation.cpp b/src/Gui/PreferencePages/DlgSettingsNavigation.cpp
index d2e54fe4e2..0d1df7fc99 100644
--- a/src/Gui/PreferencePages/DlgSettingsNavigation.cpp
+++ b/src/Gui/PreferencePages/DlgSettingsNavigation.cpp
@@ -359,7 +359,7 @@ void DlgSettingsNavigation::retranslate()
     // add submenu at the end to select navigation style
     std::map<Base::Type, std::string> styles = UserNavigationStyle::getUserFriendlyNames();
     for (const auto& style : styles) {
-        QByteArray data(style.first.getName());
+        QByteArray data(style.first.getName().data(), style.first.getName().size());
         QString name = QApplication::translate(
             std::string {style.first.getName()}.c_str(),
             style.second.c_str()
diff --git a/src/Mod/Start/Gui/GeneralSettingsWidget.cpp b/src/Mod/Start/Gui/GeneralSettingsWidget.cpp
index f87375b778..3dec7ef2a8 100644
--- a/src/Mod/Start/Gui/GeneralSettingsWidget.cpp
+++ b/src/Mod/Start/Gui/GeneralSettingsWidget.cpp
@@ -248,7 +248,7 @@ void GeneralSettingsWidget::retranslateUi()
     );
     std::map<Base::Type, std::string> styles = Gui::UserNavigationStyle::getUserFriendlyNames();
     for (const auto& style : styles) {
-        QByteArray data(style.first.getName());
+        QByteArray data(style.first.getName().data(), style.first.getName().size());
         QString name = QApplication::translate(
             std::string {style.first.getName()}.c_str(),
             style.second.c_str()
